### ISSUE TITLE
tagging old commit

### ISSUE THREAD

**Issue by larshp** [larshp/abapGit#1150]
implement feature to tag old commit SHA1

**Comment by christianguenter2** [comment]
Should we create a separate menu point like 'Create from commit' or should always select the commit like this?

![image](https://user-images.githubusercontent.com/17437789/35064155-67258ac6-fbc9-11e7-95dc-f7a6e3eb0dbc.png)

**Comment by larshp** [comment]
I was thinking about opening for input in the commit sha1 field in the current popup. Selecting from list wont scale to thousands of commits

**Comment by christianguenter2** [comment]
Am I right that currently the only way to get all commits of a branch is LCL_BRANCH_OVERVIEW=>RUN in ZABAPGIT_PAGE_BRANCH_OVERVIEW? Maybe we should refactor that nearer to the repository (e.g. LCL_REPO)?

**Comment by larshp** [comment]
yeah, but I dont think we should use it in this case, just leave it to the user to find the sha1

**Comment by christianguenter2** [comment]
hm, ok. But then we have minor inconveniences that I would like to avoid. We can't pre check the entered SHA and let the user know. He'll get a fairly non-saying error when he enters an invalid SHA.
![image](https://user-images.githubusercontent.com/17437789/35114241-60f56624-fc84-11e7-99b3-1cbf9ea2ffb8.png)

And we can't use short SHAs. AFAIS the long SHAs are not visible in abapGit and I'd like to support the short ones for convenience reasons.

Anyhow I submit a PR with the current implementation. Maybe we can adress the mentioned points sometime later.
